kube_controller_manager/datadog_checks/kube_controller_manager/kube_controller_manager.py

```diff
@@ -5,9 +5,12 @@
 from six import iteritems
 
 from datadog_checks.checks.openmetrics import OpenMetricsBaseCheck
+from datadog_checks.config import is_affirmative
 
+from datadog_checks.base.checks.kube_leader import KubeLeaderElectionMixin
 
-class KubeControllerManagerCheck(OpenMetricsBaseCheck):
+
+class KubeControllerManagerCheck(OpenMetricsBaseCheck, KubeLeaderElectionMixin):
```

Comment:
In python - at least python2 - the class hierarchy is defined right to left, so in this case the `KubeLeaderElectionMixin` class is the base class, extended by `OpenMetricsBaseCheck `. This might be fine if the mixin doesn't override any of the base class' methods. But if you do override methods or properties in your mixins this can lead to unexpected results because the priority of how methods are resolved is from left to right.

So... it's probably safer if we reverse this order: 
```class KubeControllerManagerCheck(KubeLeaderElectionMixin, OpenMetricsBaseCheck)```

Reply:
TIL thanks!